### ISSUE TITLE
Re-licensing under MIT after licensing under GPL violates the GPL

### ISSUE THREAD

**Issue by Nolij** [roblox-linux-wrapper/roblox-linux-wrapper#199]
Stated in title. I'm not a legal expert, but to the best of my knowledge licensing previously GPL code under any license other than the GPL violates the GPL, which would make the latest commit (9d44b07), which re-licenses this project under the MIT license a violation of the GPL, as it was previously GPLv3. If you want to use the MIT license (or any license other than GPL) for this project a rewrite is required.

TL;DR pretty sure 9d44b07 needs to be reverted as it violates the GPL

**Comment by Gskartwii** [comment]
What if the person choosing the license is the copyright holder themselves? It would seem to me that the copyright holder has the right to apply multiple licenses at their discretion.

**Comment by Nolij** [comment]
> What if the person choosing the license is the copyright holder themselves? It would seem to me that the copyright holder has the right to apply multiple licenses at their discretion.

As I said not a legal expert but from what I understand while it is the case with other licenses such as MIT, GPL does not allow this, as the previous code is still protected by the GPL, and therefore the newer (or in this case identical except for the license) version is violating the license under which that previous version was released, which is why re-licensing GPL code to a non-GPL license is a violation. From what I can tell this is one of the main things people don't like about the GPL. I personally like this about the GPL as it ensures that the code remains free software forever assuming the GPL is followed but I understand why that may bother some people.

**Comment by ArthurCose** [comment]
Also not a legal expert but I've seen [this](https://opensource.stackexchange.com/a/7001).

Everyone who has contributed needs to agree on the license change, and the old license can still be used by anyone with an older version of the project.

**Comment by alfonsojon** [comment]
This has been corrected, thank you for bringing this to my attention. I learned something about licensing :+1:
